Tests/XCUITests/HistoryTests.swift

```diff
@@ -57,33 +59,35 @@ class HistoryTests: BaseTestCase {
         XCTAssertTrue(app.tables.buttons["Sync and Save Data"].exists, "Sign in button does not appear")
     }
 
+    // https://testrail.stage.mozaws.net/index.php?/cases/view/2307487
     func testClearHistoryFromSettings() throws {
-        throw XCTSkip("MTE-514 Database may not be loaded")
-        /*
-        navigator.nowAt(NewTabScreen)
-
-        // Browse to have an item in history list
-        navigator.goto(LibraryPanel_History)
-        mozWaitForElementToExist(app.tables.cells[HistoryPanelA11y.recentlyClosedCell], timeout: TIMEOUT)
-        XCTAssertTrue(app.tables.cells.staticTexts[webpage["label"]!].exists)
-        XCTAssertFalse(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
-
-        // Go to Clear Data
-        navigator.goto(NewTabScreen)
-        navigator.performAction(Action.AcceptClearPrivateData)
-
-        // Back on History panel view check that there is not any item
-        navigator.goto(HomePanelsScreen)
-        navigator.goto(LibraryPanel_History)
-        mozWaitForElementToExist(app.tables[HistoryPanelA11y.tableView])
-        mozWaitForElementToExist(app.tables.cells[HistoryPanelA11y.recentlyClosedCell])
-        XCTAssertFalse(app.tables.cells.staticTexts[webpage["label"]!].exists)
-        XCTAssertTrue(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
-        */
+        XCTExpectFailure("The app was not launched", strict: false) {
+            navigator.nowAt(NewTabScreen)
+            // Browse to have an item in history list
+            navigator.goto(LibraryPanel_History)
+            mozWaitForElementToExist(app.tables.cells[HistoryPanelA11y.recentlyClosedCell], timeout: TIMEOUT)
+            XCTAssertTrue(app.tables.cells.staticTexts[oldHistoryEntries[0]].exists)
+            XCTAssertFalse(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
+
+            // Clear all private data via the settings
+            navigator.goto(HomePanelsScreen)
+            navigator.nowAt(NewTabScreen)
+            navigator.goto(ClearPrivateDataSettings)
+            app.tables.cells["ClearPrivateData"].tap()
+            app.alerts.buttons["OK"].tap()
+
+            // Back on History panel view check that there is not any item
+            navigator.goto(LibraryPanel_History)
+            mozWaitForElementToExist(app.tables[HistoryPanelA11y.tableView])
+            mozWaitForElementToExist(app.tables.cells[HistoryPanelA11y.recentlyClosedCell])
+            XCTAssertFalse(app.tables.cells.staticTexts[oldHistoryEntries[0]].exists)
+            XCTAssertTrue(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
+        }
     }
 
+    // https://testrail.stage.mozaws.net/index.php?/cases/view/2307014
     // Smoketest
-    func testClearPrivateDataButtonDisabled() throws {
```

Comment:
I also realized that this test is not working correctly, either we change the latest assertion or we modify how it works. 
Since the toggles to clear data about Browsing, HIstory and son are enabled, it is possible to click on Clear Private Data and the alert button is shown. 
We should change the assertion or be sure that all toggles are disabled before.

That said, there might be a bug because even though the toggles are enabled if there is not data to clear, should still be enabled the button to clear data? I think while that's clarified, we can just change the assertion for now

Reply:
i understan the part that this might be a bug, i agree this is a strange behaviour and needs to be clarified, but what do you think the assertion at the end should be?

Reply:
You know... better to open an issue, we can add a comment before that assertion line with // issue #... disabling this assertion until we get a clarification... wdyt?
We may need to re-do the test if the behaviour has changed... but to unblock this for now.

Reply:
Issue added, feel free to change it or add any info if you consider necessary.
https://github.com/mozilla-mobile/firefox-ios/issues/16636

Reply:
Updated.